MayssaraOmar/assignment3/Word search/src/dictionary.cpp

```diff
@@ -0,0 +1,84 @@
+#include "dictionary.h"
+
+dictionary::dictionary() {
+	root = unique_ptr<node>(new node);
+}
+void dictionary::insert(const string &word) {
+
+	node* ptr = root.get();
+	for (char c : word) {
+		if (!ptr->children[c]) {
+			ptr->children[c] = unique_ptr<node>(new node);
+		}
+		ptr = ptr->children[c].get();
+	}
+	ptr->isEndWord = true;
+	return;
```

Comment:
Optional. It is okay to have a return at the end of the function, but it is actually not required, so I would suggest leaving it out.